api.yaml

```diff
@@ -205,3 +216,109 @@ endpoints:
               statusCode: "400"
           type: http
           uri: "https://${stageVariables.enterprise_host}/enterprise/api/v1/catalogs/"
+id_parameter:
+  in: path
+  name: id
+  required: true
+  type: number
+limit_qs_parameter:
+  in: query
+  name: limit
+  required: false
+  type: number
+offset_qs_parameter:
+  in: query
+  name: offset
+  required: false
+  type: number
+page_qs_parameter:
+  in: query
+  name: page
+  required: false
+  type: number
+produces:
+  - application/json
+  - application/csv
+responses:
+  200:
+    description: OK
+  400:
+    description: "Bad Request"
+  401:
+    description: Unauthorized
+  403:
+    description: Forbidden
+  404:
+    description: "Not Found"
+  429:
+    description: "Too Many Requests"
+  500:
+    description: "Internal Server Error"
+x-amazon-apigateway-integration:
+  httpMethod: GET
+  requestParameters:
+    integration.request.header.Authorization: method.request.header.Authorization
+    integration.request.querystring.limit: method.request.querystring.limit
+    integration.request.querystring.offset: method.request.querystring.offset
+    integration.request.querystring.page: method.request.querystring.page
+  responses:
+    200:
+      statusCode: "200"
+    401:
+      statusCode: "401"
+    403:
+      statusCode: "403"
+    404:
+      statusCode: "404"
+    429:
+      statusCode: "429"
+    500:
+      statusCode: "500"
+    default:
+      statusCode: "400"
+  type: http
+x-amazon-apigateway-integration-with-id:
+  httpMethod: GET
+  requestParameters:
+    integration.request.header.Authorization: method.request.header.Authorization
+    integration.request.path.id: method.request.path.id
+  responses:
+    200:
+      statusCode: "200"
+    401:
+      statusCode: "401"
+    403:
+      statusCode: "403"
+    404:
+      statusCode: "404"
+    429:
+      statusCode: "429"
+    500:
+      statusCode: "500"
+    default:
+      statusCode: "400"
+  type: http
+x-amazon-apigateway-integration-with-id-and-querystring-parameters:
+  httpMethod: GET
+  requestParameters:
+    integration.request.header.Authorization: method.request.header.Authorization
+    integration.request.path.id: method.request.path.id
+    integration.request.querystring.limit: method.request.querystring.limit
+    integration.request.querystring.offset: method.request.querystring.offset
+    integration.request.querystring.page: method.request.querystring.page
+  responses:
+    200:
+      statusCode: "200"
+    401:
+      statusCode: "401"
+    403:
+      statusCode: "403"
+    404:
+      statusCode: "404"
+    429:
+      statusCode: "429"
+    500:
+      statusCode: "500"
+    default:
+      statusCode: "400"
+  type: http
```

Comment:
It would be good to use [aliases](https://github.com/cyklo/Bukkit-OtherBlocks/wiki/Aliases-(advanced-YAML-usage)) to reduce some of the duplication here. It totally just googled that link though so let me know if that is not a real thing.